### PR TITLE
Fix native install-dependencies.sh ref

### LINKED ISSUE #4721: VMR OSX official build legs are broken
Generating script.
Script contents:
/Users/runner/work/1/s/src/runtime/eng/install-native-dependencies.sh osx
========================== Starting Command Output ===========================
/bin/bash --noprofile --norc /Users/runner/work/_temp/23a2cf65-197c-49c5-8ace-4a9f2eb0c729.sh
/Users/runner/work/_temp/23a2cf65-197c-49c5-8ace-4a9f2eb0c729.sh: line 1: /Users/runner/work/1/s/src/runtime/eng/install-native-dependencies.sh: No such file or directory

##[error]Bash exited with code '127'.
Finishing: Install dependencies

https://github.com/dotnet/sdk/blob/e21098f6b71b2801879440a9dc687c135ea138dd/eng/pipelines/templates/jobs/vmr-build.yml#L301-L304

cc @am11 @akoeplinger

### PR DESCRIPTION
**Pull request by am11**
It was moved to arcade.
Fixes https://github.com/dotnet/source-build/issues/4721.

### CODE REVIEW

**Comment by am11** [comment]
cc @ViktorHofer

File: eng/pipelines/templates/jobs/vmr-build.yml [hunk 1]
```diff
@@ -300,7 +300,7 @@ jobs:
   - ${{ else }}:
     - ${{ if eq(parameters.targetOS, 'osx') }}:
       - script: |
-          $(sourcesPath)/src/runtime/eng/install-native-dependencies.sh osx
+          $(sourcesPath)/src/runtime/eng/common/native/install-dependencies.sh osx
```
Comment (ViktorHofer) [inline eng/pipelines/templates/jobs/vmr-build.yml:303]:
```suggestion
          $(sourcesPath)/eng/common/native/install-dependencies.sh osx
```

Comment (am11) [reply eng/pipelines/templates/jobs/vmr-build.yml:303]: Is it pointing to https://github.com/dotnet/sdk/blob/main/eng/common/native/install-dependencies.sh or https://github.com/dotnet/dotnet/blob/main/eng/common/native/install-dependencies.sh? We will need to mark it executable first. The one in runtime is executable.

Comment (ViktorHofer) [reply eng/pipelines/templates/jobs/vmr-build.yml:303]: The latter: dotnet/dotnet

Comment (ViktorHofer) [reply eng/pipelines/templates/jobs/vmr-build.yml:303]: Updating the sdk one should be sufficient to get the VMR one updated.

**Review by ViktorHofer — APPROVED** [review]
Thanks

**Review by jkoritzinsky — APPROVED** [review]

**Pull request merged**